Foundation/NSPathUtilities.swift

```diff
@@ -590,7 +590,7 @@ public func NSFullUserName() -> String {
 }
 
 internal func _NSCreateTemporaryFile(_ filePath: String) throws -> (Int32, String) {
-    let template = "." + filePath + ".tmp.XXXXXX"
```

Comment:
With the "." in place, the `fd` returned by system library call mkstemp() is always `-1` and the temporary file creation always fails. Also, since the file path sent to  _NSCreateTemporaryFile() is absolute, making it relative here does not seems right.

Reply:
The file path to `_NSCreateTemporaryFile` should not need to be absolute; we're indeed using it as a template to create a new file, so we should be able to make whatever modifications necessary internally to make it work. What path are you seeing failure for? And what's the error? We should fix the underlying problem, not the call sites which use this.